cuid2.go

```diff
@@ -74,11 +75,12 @@ func createFingerprint(random func() float64) string {
 	return hash(sourceString)[:bigLength]
 }
 
-func createCounter(start int64) func() int64 {
-	count := start
+func CreateCounter(start int64) func() int64 {
+	count := atomic.Int64{}
+	count.Store(start)
 	return func() int64 {
-		count++
-		return count
+		defer count.Add(1)
+		return count.Load()
 	}
 }
```

Comment:
How about we get rid of `differ` and `Load`. We can do something like...
```go
func CreateCounter(start int64) func() int64 {
	count := atomic.Int64{}
	count.Store(start)
	return func() int64 {
		//defer count.Add(1)
		return count.Add(1) - 1
	}
}
```

It would work exactly the same and we don't have to separate increment operation with return

Reply:
Instead of that, I would set the start with `start - 1` so that we don't have to do the calculation on each run

Reply:
That's even better!